sklearn/decomposition/_incremental_pca.py

```diff
@@ -311,7 +311,7 @@ def partial_fit(self, X, y=None, check_input=True):
         self.explained_variance_ = explained_variance[:self.n_components_]
         self.explained_variance_ratio_ = \
             explained_variance_ratio[:self.n_components_]
-        if self.n_components_ < n_features:
+        if self.n_components_ not in (n_samples, n_features):
```

Comment:
I think that a comment is required to mention that as stated by @NicolasHug: "since we already ensure that we have n_components <= (n_samples, n_features)".